spec/translator_spec.lua

```diff
@@ -0,0 +1,307 @@
+            end
+        ]],
+        [[
+            --
+            local function add(x, y)
+                return x + y
+            end
+
+            return {
+                add = add
+            }
+        ]])
+    end)
+
+    pending("Exported variables will be made local, but added to the table returned.", function ()
+        assert_translation([[
+            export x : integer = 83
+        ]],
+        [[
+            local x = 83
+
+            return {
+                x = x
+            }
+        ]])
+    end)
+
+    pending("Expressions are copied as is", function ()
+        assert_translation([[
+            export function expression()
+                local x = (1 + 2) * (100 / 30) or true
+            end
+        ]],
+        [[
+            local function expression()
+                local x = (1 + 2) * (100 / 30) or true
+            end
+
+            return {
+                expression = expression
+            }
+        ]])
+    end)
+
+    pending("While statements", function ()
+        assert_translation([[
+            local function count()
+                local i : integer = 1
+                while i <= 10 do
+                    i = i + 1
+                end
+            end
+        ]],
+        [[
+            local function count()
+                local i = 1
+                while i <= 10 do
+                    i = i + 1
+                end
+            end
+        ]])
+    end)
+
+    pending("Do Statement", function ()
+        assert_translation([[
+            local function example()
+                local i : integer = 10
+                do
+                    local i : integer = 20
+                end
+                return i
+            end
+        ]],
+        [[
+            local function example()
+                local i = 10
+                do
+                    local i = 20
+                end
+                return i
+            end
+        ]])
+    end)
+
+    pending("If statement", function ()
+        assert_translation([[
+            local function is_even(n: integer): boolean
+                if (n % 2) == 0 then
+                    return true
+                else
+                    return false
+                end
+            end
+        ]],
+        [[
+            local function is_even(n)
+                if (n % 2) == 0 then
+                    return true
+                else
+                    return false
+                end
+            end
+        ]])
+    end)
+
+    pending("For statement", function ()
+        assert_translation([[
+            local function print_strings(strings: {string})
+                for s : string in strings do
+                    io.write(s .. '\n')
+                end
+            end
+        ]],
+        [[
+            local function print_strings(strings)
+                for s in strings do
+                    io.write(s .. '\n')
+                end
+            end
+        ]])
+    end)
+end)
```

Comment:
Please add a newline here.

The weird thing is that our editorconfig is supposed to be telling the editor to add this newline. I don't know why sublime text didn't do it.

Reply:
I was not able to setup the editorconfig.

Reply:
What was the problem that you encountered? I don't use sublime text myself but the editor config website says that there is a plugin for it.

https://github.com/sindresorhus/editorconfig-sublime#readme

Reply:
I installed the plugin and luacheck. But I don't get any warnings.